packages/api/cms-api/src/warnings/decorators/create-warnings.decorator.ts

```diff
@@ -0,0 +1,38 @@
+import { type AnyEntity } from "@mikro-orm/postgresql";
+import { type CustomDecorator, SetMetadata, type Type } from "@nestjs/common";
+
+import { type WarningData } from "../dto/warning-data";
+
+interface BulkCreateWarningData {
+    warnings: WarningData[];
+    tableRowId: string;
+}
+
+export type CreateWarningsFunction<Entity extends AnyEntity = AnyEntity> = (item: Entity) => Promise<WarningData[]>;
+type CreateWarningsBulkFunction = () => AsyncGenerator<BulkCreateWarningData>;
```

Comment:
```suggestion
type BulkCreateWarningsFunction = () => AsyncGenerator<BulkCreateWarningData>;
```

Reply:
changed in: a38e2014f9bb98fa5a0e97cfdd970e2a0df07b38